### ISSUE TITLE
TriggerTemplate deletion does not clean up Taskrun's created by Triggertemplate

### ISSUE THREAD

**Issue by ibexmonj** [tektoncd/triggers#817]
# Expected Behavior

- Using tekton-trigger 0.3.1 on kubernetes version 1.14.
- On deletion of Triggertemplates I would expect Taskrun pod's that were created as part of a trigger event to be cleaned up along with the triggertemplate resource. 

# Actual Behavior

I have observed this behavior when deleting triggertemplates via helm delete as well as manual triggertemplate deletes.

# Steps to Reproduce the Problem

1. Create/Deploy an Eventlistener, Triggertemplate and Triggerbinding resource via helm chart deploy. 
2. Create a cronjob to send a trigger to the eventlistener.
3. Upon successful trigger event the triggertemplate should create a taskrun pod. 
4. Delete the helm release.
5. All the resources created as part of the helm chart are deleted successfully (cronjob, eventlistener, triggertemplate and triggerbinding) except the Taskrun pods. 

# Additional Info

- Is manual cleanup the only way to clean these taskrun pods ? 
- We expect to be running a very large workload of 1000's of jobs in our cluster and this could easily become overwhleming.
- Is this expected behavior or is this a bug ?
- FWIW i could not find a similar issue, please let me know if this has been addressed in any other issue.

Thank you again for all this community does. I have had nothing but positive things to say about my interaction with the tekton community these past couple months. Keep up the great work everyone !

**Comment by dibyom** [comment]
We do not add the trigger template as the owner of the created taskruns so this is expected behavior.  This would be a neat feature but I'm not sure if deleting all taskruns on a triggertemplate deletion would be a good default for all users (I can see some users wanting the opposite -- keeping the taskruns around).

The issue of deleting taskrun pods has come up quite a few times. For our dogfooding cluster, we have a cronjob to do this. The EventListener adds some [standard labels](https://github.com/tektoncd/triggers/blob/master/docs/eventlisteners.md#labels) that it adds to each resource it creates. One way to delete would be to delete all taskruns that match one of those labels (e.g. all from the same trigger)

**Comment by ibexmonj** [comment]
@dibyom Appreciate the response. 

Granted a vanilla delete option is not ideal. But something along  the lines of `spec.successfulJobsHistoryLimit: n` for Cronjob where it keeps `n` number of cronjobs in the history would be valuable. 

In the meanwhile, i will review the cronjob in the dogfooding repo.

**Comment by dibyom** [comment]
> Granted a vanilla delete option is not ideal. But something along the lines of spec.successfulJobsHistoryLimit: n for Cronjob where it keeps n number of cronjobs in the history would be valuable.

Totally agree that this is a very common and important use case.  We've had a few feature request for this in the pipelines repo (https://github.com/tektoncd/pipeline/issues/2856, https://github.com/tektoncd/pipeline/issues/1334, https://github.com/tektoncd/pipeline/issues/1302). I think this is something we definitely need to solve but I'm not sure if this is something that has to be part of Triggers -- it might make more sense to have this in Pipelines itself.

**Comment by ibexmonj** [comment]
@dibyom thanks for the background info and for sharing those additional issues. 

Looks like the pipeline issue has been closed and they mention `Cronjob` is the way to go there. 

Maybe this is something i can contribute to if its low key priority and no one is working on it. I just have to get familiar with how and what i need to do to buid, test, contribute etc.

**Comment by dibyom** [comment]
> Maybe this is something i can contribute to if its low key priority and no one is working on it. I just have to get familiar with how and what i need to do to buid, test, contribute etc.

Yeah, this issue keeps coming up from time to time. At some point we were considering adding a TTL controller to Pipelines itself before settling on the CronJob solution. I'd suggest starting a thread on slack or commenting on the CronJob issue above to figure out what we want to do here.
